config/application.rb

```diff
@@ -17,7 +18,14 @@
 
 module Upaya
   class Application < Rails::Application
-    AppConfig.setup(YAML.safe_load(File.read(Rails.root.join('config', 'application.yml'))))
+    configuration = AppConfigReader.new.read_configuration
+
+    # Write a copy of the current config for inspection on the instance
+    FileUtils.mkdir_p(Rails.root.join('tmp'))
+    File.write(Rails.root.join('tmp', 'application.yml'), configuration.to_yaml)
+    FileUtils.chmod(0o640, Rails.root.join('tmp', 'application.yml'))
```

Comment:
logic in this file can be hard to find and hard to test, WDYT of wrapping it up in the `AppConfigReader` class?

```suggestion
    configuration = AppConfigReader.new.read_configuration(
      write_copy_to: Rails.root.join('tmp', 'application.yml'),
    )
```

and then inside the file, something like this... `File.basename` to avoid contructing the directory path manually

```ruby
  if write_copy_to
    FileUtils.mkdir_p(File.basename(write_copy_to))
    File.write(write_copy_to, configuration.to_yaml)
    FileUtils.chmod(0o640, write_copy_to)
  end
```